bql/grammar/grammar_test.go

```diff
@@ -94,6 +94,8 @@ func TestAcceptByParse(t *testing.T) {
 		`select ?a from ?b where {?a ?p ?o} having ?b = ?b;`,
 		`select ?a from ?b where {?a ?p ?o} having (?b and ?b) or not (?b = ?b);`,
 		`select ?a from ?b where {?a ?p ?o} having ((?b and ?b) or not (?b = ?b));`,
+		`select ?a from ?b where {?a ?p ?o} having ?b = "foo"@[];`,
```

Comment:
Can you add tests with cases that should be rejected?
At TestRejectByParse

Something like:
		`select ?a from ?b where {?a ?p ?o} having ?b < "foo"@[2016-04-01T00:00:00-08:00];`
		`select ?a from ?b where {?a ?p ?o} having ?b < "foo"@[];`
		`select ?a from ?b where {?a ?p ?o} having ?b > "foo"@[2016-04-01T00:00:00-08:00];`
		`select ?a from ?b where {?a ?p ?o} having ?b > "foo"@[];`

And any other tests you may think of.

Reply:
Because of the way the grammar is structured inside `havingClauses()` and `havingClausesBinaryCompositeClauses()`, we cannot differentiate a `lexer.ItemPredicate` from a `lexer.ItemLiteral`, for example, with regards to which operations are allowed (`EQ`, `GT` and `LT`) for each of them at the grammar level.

These tests, then, are indeed accepted by parse, which is good because otherwise the error message for the user would be a parsing error message instead of the instructing one I prepared for her in the case of predicate inequalities inside `expression.go` (that instruct the user to use the `ID` and `AT` keywords to do binding extractions and use them for comparisons):
 https://github.com/google/badwolf/blob/bd3e87062c91ae981519f03c21c00e96439d4be6/bql/semantic/expression.go#L282

This way, if I were to add these tests you suggested it should be to `TestAcceptByParse` instead. What do you think?